pkgs/build-support/setup-hooks/writable-tmpdir-as-home.sh

```diff
@@ -0,0 +1,14 @@
+# shellcheck shell=bash
+# This setup hook set the HOME environment variable to a temporary directory.
+
+export HOME
+
+writableTmpDirAsHome () {
+    if [[ ! -w "$HOME" ]]; then
+        HOME=$(mktemp -d)
```

Comment:
Maybe add $HOME to a subdir of $NIX_BUILD_TOP?

This will helps to see fail build by
```
nix build --keep-failed --debug --print-build-logs .#my-pkg-name
```

Reply:
Good point, done.